docs-src/prod-readiness-context/how-to-set-up-codec-server.md

```diff
@@ -1,44 +1,245 @@
+- `Access-Control-Allow-Methods`
+- `Access-Control-Allow-Headers`
+
+For example, for Temporal Cloud Web UI hosted at https://cloud.temporal.io, enable the following in your Codec Server:
+
+- `Access-Control-Allow-Origin: https://cloud.temporal.io`
+- `Access-Control-Allow-Methods: POST, GET, OPTIONS`
+- `Access-Control-Allow-Headers: X-Namespace, Content-Type`
+
+For details on what a sample request/response looks like from the Temporal Web UI, see [Sample Request/Response](#sample-request/response).
+If setting authorization, include `Authorization` in your `Access-Control-Allow-Headers`.
+For details on setting up authorization, see [Authorization](#authorization).
+
+#### Authorization
+
+To enable authorization, your Codec Server must be an HTTPS Server.
+
+**Temporal Cloud**
+
+Temporal Cloud uses Auth0 to authenticate access.
+The Temporal Cloud UI provides an option to pass access tokens (JWT) to your Codec Server endpoints.
+Use the access tokens to validate access and then return decoded payloads from the Codec Server.
+
+You can enable this by selecting **Pass access token** in your Codec Server endpoint interface where you add your endpoint.
+Enabling this option in the Temporal Cloud UI adds an authorization header to each request sent to the Codec Server endpoint that you set.
+
+In your Codec Server implementation, verify the signature on this access token (in your authorization header) against the JWKS endpoint provided to you. <!--Is this process defined? when a customer signs up for temporal cloud, do we provide them with the JWKS as part of the onboarding process?-->
+If you want to unpack the claims in your token to add additional checks on whether the user has valid access to the namespace and payloads they are trying to access, you can implement it using Auth0 SDKs, middleware, or one of the third-party libraries at JWT.io.
+The claims in the token provided from Temporal Cloud UI contain the following detail:
+
+- `id`: Email ID of the person requesting access to the payloads.
+- `<namespace name>: <permissions>`: Namespace name and read/write permissions to the namespace. For example:
+
+  ```json
+  mynamespace:read
+  anothernamespace:write
+  ```
+
+Based on the identity information in the claims, you can set conditions in your Codec Server whether to return decoded payloads or just return the original encoded payloads.
+
+**Self-hosted Temporal Cluster**
+
+On self-hosted Temporal Clusters, configure [authorization in the Web UI configuration](/references/web-ui-configuration#auth) in your Temporal Cluster setup.
+
+With this enabled, you can pass access tokens to your Codec Server and validate the requests from the Web UI to the Codec Server endpoints that you set.
+
+Note that with self-hosted Temporal Clusters, you must explicitly configure authorization specifications for the Web UI and CLI.
+
+The following samples provide implementation examples for applying authentication on your Codec Server using the Go SDK.
```

Comment:
Change "samples provide" to "sample provides" to match the single sample listed? Or do you expect more samples Real Soon Now?